epam_python_course/L2/HW/task1_phonebook.py

```diff
@@ -0,0 +1,206 @@
+может повторятся
+       Пример:
+       имя         номер
+       Кирилл      11111111
+       Кирилл      22222222
+
+7. При выборе удаления человека, пользователь вводит имя человека. Программа ищет этого человека и
+удаляет его если он есть, если его нет - выводит на экран что человека нет в справочнике.
+
+Дополнительно
+Документирование функций обязательно
+"""
+
+SELECTION_TEXT = "\nPlease, choose action with phonebook: \nInput ADD for adding phone number \nInput PRINT for " \
+                 " display phone number \nInput DEL for deleting phone number\n"
+
+phonebook = {}
+
+
+def menu():
+    """
+    Console menu for select action and call selected function.
+    In case of incorrect input text  will be displayed error message.
+
+    :return:
+    """
+
+    add_func = "ADD"
+    output_func = "PRINT"
+    delete_func = "DEL"
+
+    current_selection = input(SELECTION_TEXT)
+
+    if current_selection == add_func:
+        input_name = input("Please, input name for adding phone number ")
+        input_number = input("Please, input phone number ")
+        add_number_by_name(input_name, input_number)
+
+    elif current_selection == output_func:
+        input_name = input("Please, input name for display phone numbers ")
+        output_numbers_by_name(input_name)
+
+    elif current_selection == delete_func:
+        input_name = input("Please, input name for deleting phone numbers ")
+        delete_by_name(input_name)
+
+    else:
+        print("Incorrect command for action. Please, input ADD or PRINT or DEL")
+
+
+def add_number_by_name(name: str, phonenumber: str):
+    """
+    Adding record in phonebook. In case when name is absent - will be added new record,
+    in case when name already exists new number will be added in list of phone number for this name
+
+    :param name: Name of man for record in phonebook
+    :type name: str
+
+    :param phonenumber: Phonenumber for record in phonebook
+    :type phonenumber: str
+    :return:
+    """
+
+    if name in phonebook:
+        phonebook[name].append(phonenumber)
+    else:
+        phonebook[name] = [phonenumber]
+
+
+def delete_by_name(name: str):
+    """
+    Deleting record in phonebook. Will be deleted all phonenumbers for input name.
+    In case if name is absent in Phonebook will be displayed a message about this.
+
+    :param name: Name of man in phonebook
+    :type name: str
+    :return:
+    """
+    if name in phonebook:
+        del phonebook[name]
+    else:
+        print(f"Man with name is {name} not found in phonebook")
+
+
+def output_numbers_by_name(name: str):
+    """
+    Display name and all phonenumbers which belong for this name.
+    Format of displaing is table
+
+
+    :param name:Name of man in phonebook
+    :type name: str
+    :return:
+    """
+
+    # имя         номер
+    # Кирилл      11111111
+    # Кирилл      22222222
+    if name in phonebook:
```

Comment:
We can write this a little simpler, but that's ok too.
for the future - look at "Return Early Pattern."